ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -73,34 +195,58 @@ static int
 S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, const char *bucketName,
   const char *authToken, const char *config, WT_FILE_SYSTEM **fileSystem)
 {
-    S3_FILE_SYSTEM *fs;
     S3_STORAGE *s3;
-    WT_CONFIG_ITEM v;
+    S3_FILE_SYSTEM *fs;
+    int ret;
+    WT_CONFIG_ITEM cacheDir;
+    std::string cacheStr;
+
     s3 = (S3_STORAGE *)storageSource;
 
     /* Mark parameters as unused for now, until implemented. */
-    UNUSED(session);
-    UNUSED(bucketName);
     UNUSED(authToken);
-    UNUSED(config);
 
     Aws::S3Crt::ClientConfiguration awsConfig;
     awsConfig.region = region;
     awsConfig.throughputTargetGbps = throughputTargetGbps;
     awsConfig.partSize = partSize;
 
+    /* Parse configuration string. */
+    ret = s3->wtApi->config_get_string(s3->wtApi, session, config, "cache_directory", &cacheDir);
+    if (ret == 0)
+        cacheStr = cacheDir.str;
+    else if (ret == WT_NOTFOUND) {
+        ret = 0;
```

Comment:
Do you need to set cacheDir/len in case of Not found. You can use ```cacheStr.empty()``` to make that check.